doc/source/user_guide/codelist.rst

```diff
@@ -0,0 +1,244 @@
+
+    .. code:: yaml
+
+       - Some Variable:
+         skip-region-aggregation: true
+
+  Setting *skip-region-aggregation* to ``true`` only skips the variable in question for
+  aggregation, if the variable is part of the provided data, it **is** used.
+
+* Any attributes which are arguments of
+  :meth:`aggregate_region() <pyam.IamDataFrame.aggregate_region>` will
+  be passed to that method. Examples include *method* and *weight*.
+
+* It is possible to rename the variable returned by the region processing using
+  a *region-aggregation* attribute, which must have a mapping of the target variable to
+  arguments of :meth:`aggregate_region() <pyam.IamDataFrame.aggregate_region>`.
+
+  This option can be used to compute several variables as part of the region-processing.
+  In the example below, the variable *Price|Carbon* is computed as a weighted average
+  using the CO2 emissions as weights, and in addition, the maximum carbon price within
+  each aggregate-region is added as a new variable *Price|Carbon (Max)*.
+
+    .. code:: yaml
+
+        - Price|Carbon:
+            unit: USD/t CO2
+            region-aggregation:
+              - Price|Carbon:
+                  weight: Emissions|CO2
+              - Price|Carbon (Max):
+                  method: max
+
+
+Guidelines and variable naming conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The variable name should adhere to the following conventions:
+
+*  A *|* (pipe) character indicates levels of hierarchy.
+*  Do not use spaces before and after the *|* character, but add a
+   space between words (e.g., *Primary Energy|Non-Biomass Renewables*).
+*  Do not use abbreviations (e.g, *PHEV*) unless strictly necessary.
+*  Do not use abbreviations of statistical operations (*min*, *max*,
+   *avg*) but always spell out the word.
+*  All words must be capitalised (except for *and*, *w/*, *w/o*, etc.).
+*  Add hierarchy levels where it might be useful in the future, e.g.,
+   use *Electric Vehicle|Plugin-Hybrid* instead of *Plugin-Hybrid
+   Electric Vehicle*.
+*  Do not include words like *Level* or *Quantity* in the variable,
+   because this should be clear from the context or unit.
+
+Region
+^^^^^^
+
+As is the case for the "variable" codelist, a region codelist must also follow a
+specific structure. 
+
+Each region **must** be part of a hierarchy, which means that the following nested list
+structure is required:
+
+.. code:: yaml
+
+   - Hierarchy 1:
+     - region 1:
+        some attribute: some value
+     - region 2
+   - Hierachy 2:
+     - ...  
+
+Attributes of the **regions**, in the above example *some attribute* of *region 1* are
+entirely optional and have no influence on the region-processing.
+
+Nonetheless, they can be very useful, examples are: ISO2/3-codes
+(https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes) or the list of countries
+included in a macro-region (i.e., a continent or large region).
```

Comment:
Saying that something is unrelated to something else can be confusing if users wouldn't even have thought that they are related in the first place...
```suggestion
Useful examples of region attriutes are: ISO2/3-codes
(https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes)
or the list of countries included in a macro-region (i.e., a continent).
```